### DISCUSSION TITLE
Ckan Solr 32k error - Why save the whole data_dict in Solr?

### DISCUSSION THREAD

**Discussion by MichaelHarmsTIB** [ckan/ckan#7405] [Q&A]
## Describe the bug
If you upload many rescources with resource description around 100 words or one resource with over 1500 words
you got the error.
I looked into the solr index and found, that the whole data_dict is saved. But why?
There are plenty of metadata-fields already in the index, to save the dict is not necessary.

![solr_data_dict](https://user-images.githubusercontent.com/49939451/220176653-f837a1af-492e-4123-95b9-1434eecb833d.png)

**Comment by amercader** [comment]
CKAN stores the validated `data_dict` to retrieve the metadata quickly in searches or `package_show` calls, avoiding parsing times.

1500 words definitely falls within the limits of what Solr can handle by default, so maybe there is some special about your setup.
* Can you paste the full Solr error you are getting?
* What solr version are you using?
* Are you using any plugins that modify the index dataset (calling `before_index()`)?

**Comment by MichaelHarmsTIB** [comment]
Ok, then is the only possible way to write a own plugin and use before_index?
My stakeholders are eager to collect all datacite metadata, including abstracts(in a scientific manner).
